Assets/Modules/Enemies/ProjectileSpawnController.cs

```diff
@@ -15,13 +15,17 @@ public class ProjectileSpawnController : MonoBehaviour
    
    private void Update()
     {
-        timer += Time.deltaTime;
-
-        if (timer > timeToShoot)
+        if(PlayerManager.Instance.isAlive) //Not sure if I should use the OnPlayerDiedEvent or use the Player Manager like this. I guess I always need a book like "canShoot" anyway, even if I use the event?
```

Comment:
You can find out which one to use by following this trick:
* When do you need to fire your logic? When the enemy dies? The answer here is no therefore the `OnPlayerDiedEvent` is not suitable. This logic needs to be fired/checked every frame when the player is alive. So your implementation works 👍

Reply:
Thanks hehe